src/robot/variables/finders.py

```diff
@@ -167,9 +167,12 @@ class EnvironmentFinder(object):
 
     def find(self, name):
         for getter in get_env_var, get_java_property:
-            value = getter(name[2:-1])
+            var_name, is_set, default_value = name.strip('%{').strip('}').partition('=')
+            value = getter(var_name)
             if value is not None:
                 return value
+            if is_set:     # in case if '' is desired default value
+                return default_value
```

Comment:
- Checking does the name contain embedded default value should be done already before the for loop.
- Similarly the default value should be returned outside the loop if neither of the getters return a value.
- Use `name[2:-1]`, not `name.strip('%{').strip('}')`. The latter strips all `%{` and `}` characters from both ends!
- `is_set` could be renamed to `has_default` or something similar.

Reply:
yeah -- not sure how i did it inside the loop :(
done.